### PR TITLE
Multi chart validation

### LINKED ISSUE #200: Refactor validation to support multiple charts
Not sure exactly what is needed here - need to dig in a bit more before starting work. Some (most?) of the refactor will be in shiny, but the `validateSettings()` function (and it's dependencies) might need tweaks too. Initial thoughts below.

- General idea is to support both "overall" and "chart" validation 
- "Overall" validation is used to update the settings Shiny page (which is shared across all charts)
- "Chart" validation is used to determine if a given chart can be drawn. 
- To simplify things, we are going require that setting metadata is the same across all chart for now (e.g. `id_col` can't be numeric in one chart and character in another chart)
- Almost certainly want to deal with #199  before diving in here.

### PR DESCRIPTION
**Pull request by jwildfire**
Merge after #265 

Fix #200 #269

### CODE REVIEW

File: R/validateSettings.R [hunk 1]
```diff
@@ -112,6 +115,25 @@ validateSettings <- function(data, settings, chart="eDish"){
   #valid=true if all checks pass, false otherwise
   settingStatus$valid <- settingStatus$checks%>%select(.data$valid)%>%unlist%>%all
   
+  #assess validity for each specified chart
+  settingStatus$charts <- list()
+  i<-1
+  for(chart in charts){
+    obj<-list(chart=chart)
```
Comment (bzkrouse) [inline R/validateSettings.R]: Just curious why you decided to structure this way...seems a little easier to extract chart-specific validation if under a named list  e.g. ``` settingStatus$charts[["edish"]] ```

Comment (jwildfire) [reply R/validateSettings.R]: Change you proposed looks good to me.

File: R/validateSettings.R [hunk 2]
```diff
@@ -104,39 +109,36 @@ validateSettings <- function(data, settings, chart="eDish"){
...
-  
+
   #assess validity for each specified chart
   settingStatus$charts <- list()
-  i<-1
   for(chart in charts){
-    obj<-list(chart=chart)
     if(settingStatus$valid){
-      obj$valid<-TRUE
+      settingStatus$charts[[chart]]<-TRUE
```
Comment (bzkrouse) [inline R/validateSettings.R]: looks good

File: inst/safetyGraphics_app/server.R [hunk 3]
```diff
@@ -83,32 +83,13 @@ observeEvent(settings_new$status(),{
...
-#     )
-
 for(chart in all_charts){
   callModule(
     module = renderChart,
     id = paste0("chart", chart),
     data = reactive(dataUpload_out$data_selected()),
     settings = reactive(settings_new$settings()),
-    valid = reactive(settings_new$status()$valid),
+    valid = reactive(settings_new$status()$charts[[chart]]),
```
Comment (bzkrouse) [inline inst/safetyGraphics_app/server.R]: is this loop working ok?  this type of code was very buggy for me before.

Comment (jwildfire) [reply inst/safetyGraphics_app/server.R]: Seems ok so far. Will keep an eye out for issues.

**Pull request merged**